File: backend/src/main/java/com/funmesseg/transportit/model/RouteMap.java
package com.funmesseg.transportit.model;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

import com.funmesseg.transportit.model.enums.EShippingState;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "routemap")
@Getter
@Setter
@NoArgsConstructor
public class RouteMap {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "routemapid")
    private Long routeMapId;

    @Column
    @Enumerated(EnumType.ORDINAL)
    private EShippingState state;

    @Column(name = "payloadvolume")
    private float payloadVolume;

    @Column(name = "startdate")
    private Date startDate;

    @Column 
    private float price;

    @JoinColumn(name = "driverid")
    @ManyToOne
    private Driver driver;

    @JoinColumn(name = "truckid")
    @ManyToOne
    private Truck truck;

    @JoinColumn(name = "trailerid")
    @ManyToOne
    private Trailer trailer;

    @OneToMany(mappedBy = "routeMap")
    private List<Package> packages;

    @JoinColumn(name = "citytoid")
    @ManyToOne
    private City cityTo;

    @JoinColumn(name = "cityfromid")
    @ManyToOne
    private City cityFrom;

    @Column
    private LocalDateTime deleted;

}

File: frontend/app/types/routeMap.ts
import { Indexable } from "../Components/Table/Table";
import { EShippingState } from "./EShippingState";
import { Driver, defaultDriver } from "./driver";
import { Package } from "./package";
import { Trailer, defaultTrailer } from "./trailer";
import { Truck, defaultTruck } from "./truck";

export interface RouteMap extends Indexable {
    routeMapid?: number;
    state: EShippingState;
    payloadVolume: number;
    startDate:Date;
    price?: number;
    driver:Driver;
    truck: Truck;
    trailer: Trailer;
    packages: Package[];
}

export const defaultRouteMap:RouteMap = {
    routeMapid: undefined,
    state: EShippingState.ORDERED,
    payloadVolume: 0,
    startDate: new Date(),
    driver: defaultDriver,
    truck: defaultTruck,
    trailer: defaultTrailer,
    packages: []
}